app/cdap/components/hydrator/components/LeftPanel/LeftPanel.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useState } from 'react';
+import ChevronLeft from '@material-ui/icons/ChevronLeft';
+import ChevronRight from '@material-ui/icons/ChevronRight';
+import Select from '@material-ui/core/Select';
+import SidePanel from 'components/hydrator/components/SidePanel/SidePanel';
+import MenuItem from '@material-ui/core/MenuItem';
+import { Button } from '@material-ui/core';
+import styled from 'styled-components';
+
+interface ILeftPanelProps {
+  onArtifactChange: (value: any) => void;
+  pluginsMap: any[];
+  selectedArtifact: any;
+  artifacts: any[];
+  itemGenericName: string;
+  groups: any[];
+  groupGenericName: string;
+  onPanelItemClick: (event: any, plugin: any) => void;
+  toggleSideBar: () => void;
+  isSideBarExpanded: boolean;
+}
+
+const StyledSelect = styled(Select)`
+  overflow: hidden;
+`;
+
+export const LeftPanel = ({
+  onArtifactChange,
+  selectedArtifact,
+  artifacts,
+  itemGenericName,
+  groups,
+  groupGenericName,
+  onPanelItemClick,
+}: ILeftPanelProps) => {
+  // angular has this saved in local storage - is this necessary?
+  const [isExpanded, setIsExpanded] = useState<boolean>(true);
+
+  return (
+    <>
+      <div className={`left-panel-wrapper ${isExpanded ? 'expanded' : ''}`}>
```

Comment:
There are a lot of things that need to be converted to make that work since a lot of the styles depend on the class left panel wrapper and expanded. This would make sense when we address all of the less tech debt.